packages/ui5-task-i18ncheck/lib/i18ncheck.js

```diff
@@ -0,0 +1,80 @@
+const utils = require("./utils");
+const log = require("@ui5/logger").getLogger("builder:customtask:i18ncheck");
+
+module.exports = async function ({ workspace, dependencies, options }) {
+
+
+    let propertyFilesP = workspace.byGlob(['**/i18n*.properties', '!**/node_modules/**'])
+    let xmlviewFilesP = workspace.byGlob(['**/*.view.xml', '!**/node_modules/**']);
+
+    let propertyFiles, xmlviewFiles
+    try {
+        [propertyFiles, xmlviewFiles] = await Promise.all([propertyFilesP, xmlviewFilesP]);
+    } catch (e) {
+        log.error(`Couldn't read files: ${e}`);
+    }
+
+    //Start all the async work concurrently
+
+    //: PromiseLike<any>[]
+    let xmlviewFileContentsP = []
+    xmlviewFiles.forEach(resource => {
+        let asyncWorkForContent = async () => {
+            options.configuration && options.configuration.debug && log.info(`Reading XML view: ${resource.getPath()} .`);
+            let fileContent = await resource.getString();
+            return utils.readI18nUsageFromXML(fileContent, resource.getPath());
+        }
+        xmlviewFileContentsP.push(asyncWorkForContent());
+    });
```

Comment:
forEach w/ async/await inside is an intuitive and nice idea, but doesn't work in practice as it creates side-effects: https://www.coreycleary.me/why-does-async-await-in-a-foreach-not-actually-await/ -> e.g. you can switch to for..of